plugins/modules/nmcli.py

```diff
@@ -2100,11 +2100,18 @@ def _compare_conn_params(self, conn_info, options):
         diff_after = dict()
 
         for key, value in options.items():
-            if not value:
+            # We can't just do `if not value` because then if there's a value
+            # of 0 specified as an integer it'll be interpreted as empty when
+            # it actually isn't.
+            if value != 0 and not value:
```

Comment:
What about empty strings and `False`? Are they OK to be skipped?

Reply:
The original purpose of this code was exactly to skip empty strings, so they're definitely OK to fix at least according to the person who wrote the code, since that was their intent, and they understand this stuff better than I do.

As for boolean False, or for that matter None, I can't speak to that other than to say that I was trying to make the smallest possible change here so for those values I preserved the previous behavior in my change. If no one has reported any issues with it, then :shrug: the behavior should remain as-is.